### PR TITLE
Add configuration for java-language-server

### PR DESCRIPTION
**Pull request by zupatisc**
I found this to be an easier setup compared to jdtls so I think it should be included here as an alternative.

### CODE REVIEW

File: lua/lspconfig/java_language_server.lua [hunk 1]
```diff
@@ -0,0 +1,20 @@
+local lspconfig = require('lspconfig')
+local configs = require('lspconfig/configs')
+
+local name = 'java_language_server'
+
+configs[name] = {
+	default_config = {
+		cmd = { '/usr/share/java/java-language-server/lang_server_linux.sh'};
```
Comment (lithammer) [inline lua/lspconfig/java_language_server.lua:8]: This `cmd` isn't portable.

Comment (zupatisc) [reply lua/lspconfig/java_language_server.lua:8]: I wasn't aware it has to be, besides where the script is installed will vary. I could add a comment about adding the script to ones PATH and change the cmd to reflect that.

Comment (mjlbach) [reply lua/lspconfig/java_language_server.lua:8]: If it's not portable, don't include cmd. See jdtls

Comment (zupatisc) [reply lua/lspconfig/java_language_server.lua]: Alright I've removed the cmd and instead added a short description on what to point it to.

**Comment by lithammer** [comment]
Please change the indentation from tabs to spaces (2).

File: lua/lspconfig/java_language_server.lua [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+		root_dir = lspconfig.util.root_pattern('build.gradle', 'pom.xml', '.git');
+		settings = {}
+	};
+	docs = {
+		description = [[
+https://github.com/georgewfraser/java-language-server
+
+Java language server
+
+Point `cmd` to `lang_server_linux.sh` or the equivalent script for MACOS/Windows provided by java-language-server
```
Comment (lithammer) [inline lua/lspconfig/java_language_server.lua:19]:
```suggestion
Point `cmd` to `lang_server_linux.sh` or the equivalent script for macOS/Windows provided by java-language-server
```

**Comment by lithammer** [comment]
Looks good. I'll merge after you squash the commits.

**Comment by zupatisc** [comment]
Not sure if you are notified so just a quick tag to let you know I squashed the commits @lithammer

**Comment by lithammer** [comment]
Thanks!

**Pull request merged**